test/Integration/IntegrationTestBase.cs

```diff
@@ -188,26 +188,33 @@ protected void StartFunctionHost(string functionName, SupportedLanguages languag
             {
                 StartInfo = startInfo
             };
+
+            // Register all handlers before starting the functions host process.
+            var taskCompletionSource = new TaskCompletionSource<bool>();
             this.FunctionHost.OutputDataReceived += this.TestOutputHandler;
+            this.FunctionHost.OutputDataReceived += SignalStartupHandler;
+            this.FunctionHost.OutputDataReceived += customOutputHandler;
+
             this.FunctionHost.ErrorDataReceived += this.TestOutputHandler;
 
             this.FunctionHost.Start();
             this.FunctionHost.BeginOutputReadLine();
             this.FunctionHost.BeginErrorReadLine();
 
-            var taskCompletionSource = new TaskCompletionSource<bool>();
-            this.FunctionHost.OutputDataReceived += (object sender, DataReceivedEventArgs e) =>
+            this.TestOutput.WriteLine($"Waiting for Azure Function host to start...");
+            taskCompletionSource.Task.Wait(60000);
```

Comment:
No magic numbers, make this a constant value with a descriptive name

Reply:
This line already existed earlier. I do not understand what is about this number. It's just like any other literal value like `true` or `Common.SupportedLanguages.CSharp`. It simply says wait until 60,000 milliseconds.

Reply:
https://en.wikipedia.org/wiki/Magic_number_(programming)

`Unique values with unexplained meaning or multiple occurrences which could (preferably) be replaced with named constants`

In this case is the meaning potentially clear enough? Probably. But in our code we want to be as clear as possible and adding a const with a useful name instead of hardcoded numbers makes it easier to understand and helps encourage this pattern across the code base.

Reply:
(the fact that it existed like this earlier was likely just an oversight or code that we inherited. I'd appreciate it if you could just change that now since you're making changes here anyways)

Reply:
I can make the change as it is nonconsequential. I think the line is blurred between what numbers can and cannot be treated as magic. In my opinion, a magic number would be something like `56764` instead of `60000` that will tend to perplex the reader and make it difficult for him/her decide why that exact number. Here, these are ballpark numbers to give ample time for the host does not come up. I am not sure whether moving these values to local variable for one-time use will make the improve the readability. I find the change to be similar to replacing:
```cs
this.StartFunctionHost(nameof(ProductTrigger), Common.SupportedLanguages.CSharp, true);
```
with
```cs
string functionName = nameof(ProductTrigger);
SupportedLanguages language = Common.SupportedLanguages.CSharp;
bool useTestFolder = true;

this.StartFunctionHost(functionName, language, useTestFolder);
```